### PR TITLE
[IE CLDNN] Add asymmetric dw convolution improvements

### PR DESCRIPTION
**Pull request by kdobros**
This change adds full support for asymmetric quantization to optimized
depthwise convolution, adds slm optimization and other minor
improvements.

Issue: CVS-25122

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp [hunk 1]
```diff
@@ -123,21 +111,137 @@ ConvolutionKernel_b_fs_yx_fsv_16_32_imad_dw::GetAutoTuneParams(const convolution
...
-        tune_params.tile_x = std::min((size_t)16, params.output.X().v);
+
+    auto& output = params.output;
+
+    size_t fsv = 1;
+    if (output.GetLayout() == DataLayout::b_fs_yx_fsv16) {
+        fsv = 16;
     } else {
-        tune_params.tile_x = std::min((size_t)8, params.output.X().v);
+        fsv = 32;
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp:121]: I think that you can change initialization to "size_t fsv = 32" and to remove "else" branch after that. Looks like you use only fsv = 16 and fsv = 32, so fsv = 1 don't really needed.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp [hunk 2]
```diff
@@ -123,21 +111,137 @@ ConvolutionKernel_b_fs_yx_fsv_16_32_imad_dw::GetAutoTuneParams(const convolution
...
+        // For best hw utilization work-group size should be multiple of 2, so if y isn't force it in x
+        if (lws1 % 2 != 0) {
+            lws0 = 2;
+        }
+        // Local memory allocation works at 1kb granularity
+        auto calc_slm_size_kb = [&](size_t tile_x, size_t lws0, size_t lws1) {
+            size_t slm_x = (tile_x * lws0 - 1) * params.stride.x + (params.filterSize.x - 1) * params.dilation.x;
+            size_t slm_y = (lws1 == 1) ? params.filterSize.x
+                                       : (lws1 - 1) * params.stride.y + (params.filterSize.y - 1) * params.dilation.y + 1;
+            return CeilDiv(slm_x * slm_y * fsv, 1024);
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp:177]: As I know minimum SLM allocation size is 4k per workgroup. I amn't sure but maybe it makes sence to check this limitation in your code?

return slm_x * slm_y * fsv < 4096 ? 4096 : CeilDiv(slm_x * slm_y * fsv, 1024);

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_b_fs_yx_fsv_16_32_imad_dw.cl [hunk 3]
```diff
@@ -182,20 +184,119 @@ KERNEL(convolution)(
...
+        for (; in_s < slm_total_size; in_s += LWS0 * LWS1 * SIMD * iteration_preload_bytes) {
+            uint input_f = in_s % FSV;
+            uint input_x = in_s / FSV % wg_in_tile_x;
+            uint input_y = in_s / FSV / wg_in_tile_x;
+            if (LWS1 == 1)
+                input_y *= DILATION_SIZE_Y;
+            uint input_idx = input_offset_base + input_f + input_x * INPUT_X_PITCH + input_y * INPUT_Y_PITCH;
+
+            int pad_x = wg_x * STRIDE_SIZE_X + (int)input_x < PADDING_SIZE_X || wg_x * STRIDE_SIZE_X + (int)input_x >= INPUT0_SIZE_X + PADDING_SIZE_X;
+            int pad_y = wg_y * STRIDE_SIZE_Y + (int)input_y < PADDING_SIZE_Y || wg_y * STRIDE_SIZE_Y + (int)input_y >= INPUT0_SIZE_Y + PADDING_SIZE_Y;
```
[the remark this answers is not in the document]
Comment (kdobros) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_b_fs_yx_fsv_16_32_imad_dw.cl:243]: Done

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp (hunk 1 above)
Comment (kdobros) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp]: Changed

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp (hunk 2 above)
Comment (kdobros) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp:175]: That's not exactly true, there is no problem with allocating 1kb of slm.
This is just easy explanation why we can't have 64 work-groups using 1kb slm each - we can have only up to 16 work-groups executing concurrently.
I actually approximate more detailed tradeoff between slm usage and number of threads in 
```
            bool c_slm_size = calc_slm_size_kb(c_tile, lws0, lws1) <= lws0 * lws1;
```
and by ensuring at least 4 sub-groups per work-group are used.

Comment (lznamens) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp:175]: Thank you for a good explanation.

**Pull request merged**